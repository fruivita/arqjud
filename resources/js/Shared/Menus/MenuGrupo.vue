<!--
    Agrupamento de elementos do menu principal.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import Icone from '@/Shared/Icones/Icone.vue';
import { ref } from 'vue';

defineProps({
    texto: { type: String, required: true },
});

const exibir = ref(false);
</script>

<template>
    <div class="space-y-1">
        <button
            @click="exibir = !exibir"
            class="flex w-full items-center justify-between rounded border-primaria-700 bg-primaria-100 p-2 text-left font-extrabold outline-none transition hover:border-l-4 focus:border-l-4 dark:bg-secundaria-600"
            type="button"
            dusk="btn-menu-grupo"
        >
            <span>{{ texto }}</span>

            <Icone
                nome="caret-right"
                class="h-6 w-6 transition duration-300"
                :class="{ 'rotate-90': exibir }"
            />
        </button>

        <Transition
            enter-from-class="opacity-0"
            enter-to-class="opacity-100"
            enter-active-class="transition duration-300 transform-gpu"
            leave-active-class="transition duration-200 transform-gpu"
            leave-from-class="opacity-100"
            leave-to-class="opacity-0"
        >
            <div
                v-show="exibir"
                class="flex flex-col space-y-1 border-l-2 border-primaria-500 pl-3 dark:border-secundaria-500"
                dusk="slot"
            >
                <slot></slot>
            </div>
        </Transition>
    </div>
</template>
