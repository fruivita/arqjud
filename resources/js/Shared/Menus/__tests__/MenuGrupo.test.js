/**
 * Testes para o componente MenuGrupo.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import MenuGrupo from '@/Shared/Menus/MenuGrupo.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(MenuGrupo, { ...options });
    };
});

describe('MenuGrupo', () => {
    // Caminho feliz
    test('propriedades do componente estão definidas', () => {
        expect(MenuGrupo.props).toMatchObject({
            ativo: { type: Boolean, default: false },
            texto: { type: String, required: true },
        });
    });

    test('atributos e slots do componente renderizados respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { texto: 'foo' },
                slots: { default: 'bar' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente sem estar no estado ativo (não selecionado) respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { ativo: false, texto: 'foo' },
                slots: { default: 'bar' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente com o estado ativo (selecionado) respeitando o snapshot', () => {
        expect(
            mountFunction({
                props: { ativo: true, texto: 'foo' },
                slots: { default: 'bar' },
            }).html()
        ).toMatchSnapshot();
    });

    test('altera a visibilidade do slot e disposição do ícone ao clicar no botão respeitando o snapshot', async () => {
        const wrapper = mountFunction({
            props: { texto: 'foo' },
            slots: { default: 'bar' },
        });

        expect(wrapper.get('[dusk="slot"]').isVisible()).toBeFalsy();
        expect(wrapper.html()).toMatchSnapshot();

        await wrapper.get('[dusk="btn-menu-grupo"]').trigger('click');

        expect(wrapper.get('[dusk="slot"]').isVisible()).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
    });
});
