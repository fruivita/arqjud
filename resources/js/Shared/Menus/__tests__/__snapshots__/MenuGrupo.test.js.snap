// Vitest Snapshot v1

exports[`MenuGrupo > altera a visibilidade do slot e disposição do ícone ao clicar no botão respeitando o snapshot 1`] = `
"<div class=\\"space-y-1\\"><button class=\\"bg-primaria-100 hover:border-l-4 focus:border-l-4 dark:bg-secundaria-600 flex w-full items-center justify-between rounded border-primaria-700 p-2 text-left font-extrabold outline-none transition\\" type=\\"button\\" dusk=\\"btn-menu-grupo\\"><span>foo</span><svg class=\\"h-6 w-6 transition duration-300\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M6 12.796V3.204L11.481 8 6 12.796zm.659.753 5.48-4.796a1 1 0 0 0 0-1.506L6.66 2.451C6.011 1.885 5 2.345 5 3.204v9.592a1 1 0 0 0 1.659.753z\\"></path>
    </svg></button>
  <transition-stub enterfromclass=\\"opacity-0\\" entertoclass=\\"opacity-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100\\" leavetoclass=\\"opacity-0\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
    <div class=\\"flex flex-col space-y-1 border-l-2 border-primaria-500 pl-3 dark:border-secundaria-500\\" dusk=\\"slot\\" style=\\"display: none;\\">bar</div>
  </transition-stub>
</div>"
`;

exports[`MenuGrupo > altera a visibilidade do slot e disposição do ícone ao clicar no botão respeitando o snapshot 2`] = `
"<div class=\\"space-y-1\\"><button class=\\"bg-primaria-100 hover:border-l-4 focus:border-l-4 dark:bg-secundaria-600 flex w-full items-center justify-between rounded border-primaria-700 p-2 text-left font-extrabold outline-none transition\\" type=\\"button\\" dusk=\\"btn-menu-grupo\\"><span>foo</span><svg class=\\"h-6 w-6 transition duration-300 rotate-90\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M6 12.796V3.204L11.481 8 6 12.796zm.659.753 5.48-4.796a1 1 0 0 0 0-1.506L6.66 2.451C6.011 1.885 5 2.345 5 3.204v9.592a1 1 0 0 0 1.659.753z\\"></path>
    </svg></button>
  <transition-stub enterfromclass=\\"opacity-0\\" entertoclass=\\"opacity-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100\\" leavetoclass=\\"opacity-0\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
    <div class=\\"flex flex-col space-y-1 border-l-2 border-primaria-500 pl-3 dark:border-secundaria-500\\" dusk=\\"slot\\" style=\\"\\">bar</div>
  </transition-stub>
</div>"
`;

exports[`MenuGrupo > atributos e slots do componente renderizados respeitando o snapshot 1`] = `
"<div class=\\"space-y-1\\"><button class=\\"bg-primaria-100 hover:border-l-4 focus:border-l-4 dark:bg-secundaria-600 flex w-full items-center justify-between rounded border-primaria-700 p-2 text-left font-extrabold outline-none transition\\" type=\\"button\\" dusk=\\"btn-menu-grupo\\"><span>foo</span><svg class=\\"h-6 w-6 transition duration-300\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M6 12.796V3.204L11.481 8 6 12.796zm.659.753 5.48-4.796a1 1 0 0 0 0-1.506L6.66 2.451C6.011 1.885 5 2.345 5 3.204v9.592a1 1 0 0 0 1.659.753z\\"></path>
    </svg></button>
  <transition-stub enterfromclass=\\"opacity-0\\" entertoclass=\\"opacity-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100\\" leavetoclass=\\"opacity-0\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
    <div class=\\"flex flex-col space-y-1 border-l-2 border-primaria-500 pl-3 dark:border-secundaria-500\\" dusk=\\"slot\\" style=\\"display: none;\\">bar</div>
  </transition-stub>
</div>"
`;

exports[`MenuGrupo > renderiza o componente com o estado ativo (selecionado) respeitando o snapshot 1`] = `
"<div class=\\"space-y-1\\"><button class=\\"border-l-4 bg-primaria-400 text-primaria-50 dark:bg-secundaria-800 flex w-full items-center justify-between rounded border-primaria-700 p-2 text-left font-extrabold outline-none transition\\" type=\\"button\\" dusk=\\"btn-menu-grupo\\"><span>foo</span><svg class=\\"h-6 w-6 transition duration-300\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M6 12.796V3.204L11.481 8 6 12.796zm.659.753 5.48-4.796a1 1 0 0 0 0-1.506L6.66 2.451C6.011 1.885 5 2.345 5 3.204v9.592a1 1 0 0 0 1.659.753z\\"></path>
    </svg></button>
  <transition-stub enterfromclass=\\"opacity-0\\" entertoclass=\\"opacity-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100\\" leavetoclass=\\"opacity-0\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
    <div class=\\"flex flex-col space-y-1 border-l-2 border-primaria-500 pl-3 dark:border-secundaria-500\\" dusk=\\"slot\\" style=\\"display: none;\\">bar</div>
  </transition-stub>
</div>"
`;

exports[`MenuGrupo > renderiza o componente sem estar no estado ativo (não selecionado) respeitando o snapshot 1`] = `
"<div class=\\"space-y-1\\"><button class=\\"bg-primaria-100 hover:border-l-4 focus:border-l-4 dark:bg-secundaria-600 flex w-full items-center justify-between rounded border-primaria-700 p-2 text-left font-extrabold outline-none transition\\" type=\\"button\\" dusk=\\"btn-menu-grupo\\"><span>foo</span><svg class=\\"h-6 w-6 transition duration-300\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <path d=\\"M6 12.796V3.204L11.481 8 6 12.796zm.659.753 5.48-4.796a1 1 0 0 0 0-1.506L6.66 2.451C6.011 1.885 5 2.345 5 3.204v9.592a1 1 0 0 0 1.659.753z\\"></path>
    </svg></button>
  <transition-stub enterfromclass=\\"opacity-0\\" entertoclass=\\"opacity-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100\\" leavetoclass=\\"opacity-0\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
    <div class=\\"flex flex-col space-y-1 border-l-2 border-primaria-500 pl-3 dark:border-secundaria-500\\" dusk=\\"slot\\" style=\\"display: none;\\">bar</div>
  </transition-stub>
</div>"
`;
